### PR TITLE
Adds function to enable 3DS support

### PR DESCRIPTION
**Pull request by igorp1**
__What__

When using the card form component to passthrough cards to suppliers, merchants might need to present users with a 3DS challenge. This PR introduces a function `createThreeDSecureSession` which will return a 3DS session. That session id should be used in the payment object when creating orders and bookings.

### CODE REVIEW

**Review by andrejak — DISMISSED** [review]
Great work

File: examples/card-with-3ds/server.mjs [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+  return data.component_client_key;
+}
+
+async function createCard() {
+  const response = await fetch(`${process.env.TOKEN_PROXY_URL}/vault/cards`, {
+    method: "POST",
+    headers: API_HEADERS,
+    body: JSON.stringify({
+      data: {
+        number: "4111110148486405", // BA: 4263970000005262
```
Comment (angelcarballo) [inline examples/card-with-3ds/server.mjs:68]: Where did this number come from? Just curious

File: examples/card-with-3ds/server.mjs [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+      console.error(error);
+      response.writeHead(500);
+      response.end(error.message);
+    }
+  };
+}
+
+const SERVER_ROUTES = {
+  "/": endpoint(async function (_, response) {
+    const offer = await getOffer("LHR", "ABZ", "british_airways");
```
Comment (angelcarballo) [inline examples/card-with-3ds/server.mjs:189]: Should we use Duffel Airways instead?

File: src/functions/createThreeDSecureSession/client.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  id: string;
+  live_mode: boolean;
+  resource_id: string;
+  cardholder_present: boolean;
+  status: ThreeDSSessionStatus;
+  external_id: string | null;
+}
+
+const getAPIHeaders = (clientKey: string) => ({
+  "Duffel-Version": "v1",
```
Comment (angelcarballo) [inline src/functions/createThreeDSecureSession/client.ts:17]: This should be `v2`, right?

**Review by angelcarballo — DISMISSED** [review]
Still testing, but handling of each scenario looks good 👍🏻

File: examples/card-with-3ds/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# card with 3DS example
+
+This example will demonstarte how to use the `createThreeDSecureSession` function.
```
Comment (andrejak) [inline examples/card-with-3ds/README.md:3]:
```suggestion
This example will demonstrate how to use the `createThreeDSecureSession` function.
```

File: examples/card-with-3ds/server.mjs (hunk 1 above)
Comment (andrejak) [reply examples/card-with-3ds/server.mjs:68]: There are some test numbers on here: https://docs.evervault.com/payments/3d-secure/test-your-implementation
This was one for a frictionless flow. Maybe we add a link to that page in a comment above the example number?

File: examples/card-with-3ds/README.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+The server will get an offer, tokenise a card and create a client key.
+These values will be aailable on the client to call the function.
```
[the remark this answers is not in the document]
Comment (igorp1) [reply examples/card-with-3ds/README.md:5]: ✓ fixed

File: examples/card-with-3ds/index.html [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+
+    <title>Card Form integration example</title>
+  </head>
+  <body style="height: 100vh">
+    <h1>3DS example</h1>
+
+    <p>
+      In this example we tokenise the card in the backend so we cas solely focus
```
[the remark this answers is not in the document]
Comment (igorp1) [reply examples/card-with-3ds/index.html:12]: ✓ fixed

File: examples/card-with-3ds/server.mjs (hunk 2 above)
Comment (igorp1) [reply examples/card-with-3ds/server.mjs:189]: Good shout, done! Left a BA example in there too for convenience.

File: src/functions/createThreeDSecureSession/client.ts (hunk 3 above)
Comment (igorp1) [reply src/functions/createThreeDSecureSession/client.ts:17]: It should, yes. Updated!

File: examples/card-with-3ds/README.md (hunk 4 above)
Comment (igorp1) [reply examples/card-with-3ds/README.md:3]: ✓ fixed

**Review by andrejak — DISMISSED** [review]

**Review by angelcarballo — APPROVED** [review]
Looking good 🚀

**Pull request merged**